src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -67,8 +68,8 @@ public static unsafe nuint GetIndexOfFirstNonAsciiByte(byte* pBuffer, nuint buff
             // pmovmskb which we know are optimized, and (b) we can avoid downclocking the processor while
             // this method is running.
 
-            return (Sse2.IsSupported)
-                ? GetIndexOfFirstNonAsciiByte_Sse2(pBuffer, bufferLength)
+            return (Sse2.IsSupported || AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian)
```

Comment:
Considering `&&` has precedence over `||`, is this check correct? You don't care to check for endianness in `Sse2`?

I ask because you have this in line 227:
```cs
 Debug.Assert(BitConverter.IsLittleEndian, "SSE2/Arm assumes little-endian.");
```

The same question applies for line 524.

Reply:
So in all of the existing code, we've just assumed that whenever `Sse2.IsSupported` returns true we're in a `LittleEndian` environment. I actually assumed that `Sse2.IsSupported` would check for `IsLittleEndian` internally. Now that I think about it, it'd be safer to do what you suggest. I'll make the change

Reply:
It's correct, but it was pointed out in other PRs that we might want to parenthesize for clarity.

Reply:
Done.